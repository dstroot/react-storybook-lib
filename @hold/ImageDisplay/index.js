import React from "react";
import FbImageLibrary from "react-fb-image-grid";
import PropTypes from "prop-types";

/** 
This component requires an external library. 
NOTE: `yarn add react-fb-image-grid`
 */
const ImageDisplay = ({ images }) => {
  return <FbImageLibrary hideOverlay images={images} />;
};

ImageDisplay.propTypes = {
  images: PropTypes.array.isRequired
};

export default ImageDisplay;
