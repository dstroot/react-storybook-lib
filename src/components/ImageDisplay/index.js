// yarn add react-fb-image-grid

import React from 'react';
import FbImageLibrary from 'react-fb-image-grid';
// import PropTypes from 'prop-types';

// images
import img1 from './media/1.jpg';
import img2 from './media/2.jpg';
import img3 from './media/3.jpg';
import img4 from './media/4.webp';
import img5 from './media/5.webp';
import img6 from './media/6.jpg';
import img7 from './media/7.jpg';
import img8 from './media/home.jpg';
import img9 from './media/startup.jpg';

// image array
const images = [img1, img2, img3, img4, img5, img6, img7, img8, img9];

const ImageDisplay = () => {
  return <FbImageLibrary hideOverlay images={images} />;
};

// ImageDisplay.propTypes = {
//   image: PropTypes.array.isRequired,
// };

export default ImageDisplay;
