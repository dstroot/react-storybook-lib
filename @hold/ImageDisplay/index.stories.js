import React from "react";
import { storiesOf } from "@storybook/react";

// components
import ImageDisplay from "../ImageDisplay";

// images
import img1 from "./media/1.jpg";
import img2 from "./media/2.jpg";
import img3 from "./media/3.jpg";
import img4 from "./media/4.webp";
import img5 from "./media/5.webp";
import img6 from "./media/6.jpg";
import img7 from "./media/7.jpg";
import img8 from "./media/home.jpg";
import img9 from "./media/startup.jpg";

// image array
const images = [img1, img2, img3, img4, img5, img6, img7, img8, img9];

// ImageDisplay section
storiesOf("ImageDisplay", module).add("show ImageDisplay", () => (
  <ImageDisplay images={images} />
));
