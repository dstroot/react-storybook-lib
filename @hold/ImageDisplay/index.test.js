import React from "react";
import renderer from "react-test-renderer";

import ImageDisplay from "../ImageDisplay";

// images
import img1 from "./media/1.jpg";
import img2 from "./media/2.jpg";
import img3 from "./media/3.jpg";
import img4 from "./media/4.webp";
import img5 from "./media/5.webp";
import img6 from "./media/6.jpg";
import img7 from "./media/7.jpg";
import img8 from "./media/home.jpg";
import img9 from "./media/startup.jpg";

// image array
const images = [img1, img2, img3, img4, img5, img6, img7, img8, img9];

test("ImageDisplay renders some content", () => {
  const component = renderer.create(<ImageDisplay images={images} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
