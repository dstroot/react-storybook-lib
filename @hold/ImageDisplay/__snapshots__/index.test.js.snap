// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDisplay renders some content 1`] = `
<div
  className="grid-container"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="border height-two background col-md-6 col-xs-6"
        onClick={[Function]}
        style={
          Object {
            "background": "url(1.jpg)",
          }
        }
      />
      <div
        className="border height-two background col-md-6 col-xs-6"
        onClick={[Function]}
        style={
          Object {
            "background": "url(2.jpg)",
          }
        }
      />
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="border height-three background col-md-4 col-xs-6"
        onClick={[Function]}
        style={
          Object {
            "background": "url(3.jpg)",
          }
        }
      />
      <div
        className="border height-three background col-md-4 col-xs-6"
        onClick={[Function]}
        style={
          Object {
            "background": "url(4.webp)",
          }
        }
      />
      <div
        className="border height-three background col-md-4 col-xs-6"
        onClick={[Function]}
        style={
          Object {
            "background": "url(5.webp)",
          }
        }
      >
        <div
          className="cover"
        />
        <div
          className="cover-text"
          style={
            Object {
              "fontSize": "200%",
            }
          }
        >
          <p>
            +
            4
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
